src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```
@@ -1,25 +1,69 @@
+        products.add(new Product(9L, "nokia3310", "Nokia 3310", new BigDecimal(70), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Nokia/Nokia%203310.jpg"));
+        products.add(new Product(10L, "palmp", "Palm Pixi", new BigDecimal(170), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Palm/Palm%20Pixi.jpg"));
+        products.add(new Product(11L, "simc56", "Siemens C56", new BigDecimal(70), usd, 20, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C56.jpg"));
+        products.add(new Product(12L, "simc61", "Siemens C61", new BigDecimal(80), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C61.jpg"));
+        products.add(new Product(13L, "simsxg75", "Siemens SXG75", new BigDecimal(150), usd, 40, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg"));
+    }
+
     @Override
     public Product getProduct(Long id) {
-        throw new RuntimeException("Not implemented");
+        return products.stream()
+                .filter(s->s.getId()==id && s.getPrice()!=null && s.getStock()>0)
+                .findAny()
+                .orElse(null);
     }
 
     @Override
     public List<Product> findProducts() {
-        throw new RuntimeException("Not implemented");
+        return products.stream()
+                .filter(s-> s.getPrice()!=null && s.getStock()>0)
+                .collect(Collectors.toList());
     }
 
     @Override
     public void save(Product product) {
-        throw new RuntimeException("Not implemented");
+        if (product!=null && !products.contains(product))
```

What happens if there is already such product in the list? Should we throw any exception in this case?

---

Consider rewriting using streams